src/main/java/pl/touk/repository/InMemoryDriverRepository.java

```diff
@@ -0,0 +1,62 @@
+package pl.touk.repository;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Repository;
+import pl.touk.model.Driver;
+
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.stream.Collectors;
+
+@Slf4j
+@Repository
+public class InMemoryDriverRepository implements DriverRepository {
+
+    private AtomicLong idGenerator = new AtomicLong();
+    private Map<Long, Driver> drivers = new ConcurrentHashMap<>();
+
+    @Override
+    public Driver save(Driver driver) {
+        drivers.put(idGenerator.getAndIncrement(), driver);
+        driver.setId(idGenerator.getAndDecrement());
```

Comment:
a co się stanie jak pomiędzy linią 26 a 27 ktoś inny wywoła save?

Reply:
Będzie to wywołanie rekurencyjne (błędne) i otrzymamy 'StackOverflowError'.